swimlane/core/resources/record.py

```diff
@@ -29,6 +29,8 @@ def __init__(self, app, raw):
 
         self.__app = app
 
+        self.__pending = {}
```

Comment:
Can you explain purpose of the new field? I don't see any use of it, just assignments.

Reply:
applies any pending changes to the values dictionary on the python record to `values` on the PATCH body here: https://github.com/swimlane/swimlane-python/pull/68/files#diff-bc0dbe0a004bc6c0de1acd0f4adb9900R221

Reply:
But why they are pending? Shouldn't all that you created on record go to patch or save, whatever the customer chooses? You are setting field at the same time, so why do we need duplication is a question.

Reply:
Anything we've changed since pulling the record via the GET call gets sent with POST / PUT on `save` and PATCH on `patch` but the idea here is that we don't want to send every existing, unchanged value on PATCH which is what we'd have without a separate set of `pending` changes no?

Reply:
Not from my understanding. The statement itself is correct, but the only time they are planning to use Patch is when they create partial record, with just few fields and they don't want to override existing fields. The existing records with modifications are intended to go through the Save. On top of everything it will not work, if they setup the whole collection instead of going through the individual setter.

Reply:
This is the functionality was requested by Michael Butler in the original ticket https://swimlane.atlassian.net/browse/SPT-3918 and in the discussions on the JIRA for this ticket (https://swimlane.atlassian.net/browse/SPT-5997) for the purposes of avoiding regularly overriding existing fields with calls to `save` by offering the option to call `patch` instead and only send those fields that have changed since the last "update". `save` isn't changing, I'm not sure what the issue is here.

Reply:
That is exactly what I am talking about. Partial records.

Reply:
The issue is unnecessary fields and duplication of data, plus an unexpected behavior if values collection has been setup directly i.e not through _set_item.

Reply:
As far as I'm aware we don't advertise support for changing the values collection directly via `._raw['values']`. 

I'm not sure how we support the PATCH endpoint here otherwise unless it's sending over a new collection like `.patch({'fieldA': 'a', 'fieldB': 'b'})` which was ultimately not what the request from the TAMs was.

Reply:
Should we maybe change `pending` to store an array of ids of the fields that were set since the last save/patch, rather than the values too? Then we could reconstruct `copy_raw['values']` to only include those the fields with those ids? Would that help us not have to keep the values twice?

Reply:
👍 updated to use dictionary comprehension (TIL) instead of pending values

Reply:
@zbartl While adding pending takes care of deleting fields (which was not done before) the code is still unnecessary - you can remove it and it will still function as intended. You don't need to send no new collections either, we are expecting partial record to have partial values. 
While we do not advertise modifying raw, we are constructing Record from raw, which is exactly what customer wanted to submit partial raw record. That will not go through the setter. As I was saying too many unexpected results from the code that is not needed.
Which brings me to another point what we advertise for the customers is an adapter, which has create method, what I imagine need to be done is to create patch method - which will create partial record which makes tracking absolutely unnecessary.

Reply:
Again, that’s not the functionality that was requested by the field. The patch method already exists in this PR but the record adapter isn’t tied to any 1 record and manually constructing partial records doesn’t give users a simple route to avoid the sort of workflow race conditions described by SPT-3918.

Reply:
It is exactly the functionality that was requested. The ability to submit and construct partial records. I will explain it on the meeting.